ConsoleApp/ConsoleApp/Program.cs

```diff
@@ -0,0 +1,176 @@
+                    Distance = distance
+                };
+            })
+            .ToArray();
+
+        var min = distances.MinBy(x => x.Distance);
+        var max = distances.MaxBy(x => x.Distance);
+
+        Console.WriteLine($"Min distance is between {min.First} and {min.Second} is {min.Distance} km");
+        Console.WriteLine($"Max distance is between {max.First} and {max.Second} is {max.Distance} km");
+    }
+
+    public static void MaxDistanseFromCardinalDirection(this IEnumerable<Person> persons)
+    {
+        var distances = persons.Select(person =>
+        {
+            var distanceToNorth = DistanceOnEarth(person.Latitude, person.Longitude, 90, person.Longitude);
+            var distanceToSouth = DistanceOnEarth(person.Latitude, person.Longitude, -90, person.Longitude);
+            var distanceToEast = DistanceOnEarth(person.Latitude, person.Longitude, person.Latitude, 0);
+            var distanceToWest = DistanceOnEarth(person.Latitude, person.Longitude, person.Latitude, -180);
+
+            return new
+            {
+                Name = person.Name,
+                DistanceToNorth = distanceToNorth,
+                DistanceToSouth = distanceToSouth,
+                DistanceToEast = distanceToEast,
+                DistanceToWest = distanceToWest
+            };
+        }).ToArray();
+
+        var maxFromNorth = distances.MaxBy(x => x.DistanceToNorth);
+        var maxFromSouth = distances.MaxBy(x => x.DistanceToSouth);
+        var maxFromEast = distances.MaxBy(x => x.DistanceToEast);
+        var maxFromWest = distances.MaxBy(x => x.DistanceToWest);
+
+        Console.WriteLine($"Max distance to North by {maxFromNorth.Name}, {maxFromNorth.DistanceToNorth} km");
+        Console.WriteLine($"Max distance to North by {maxFromSouth.Name}, {maxFromSouth.DistanceToSouth} km");
+        Console.WriteLine($"Max distance to North by {maxFromEast.Name}, {maxFromEast.DistanceToEast} km");
+        Console.WriteLine($"Max distance to North by {maxFromWest.Name}, {maxFromWest.DistanceToWest} km");
+    }
+
+
+    public static void SameWordsInAbout(this IEnumerable<Person> persons)
+    {
+        var group = persons.Join(persons,
+            person => true,
+            person => true,
+            (person1, person2) => new
+            {
+                First = person1,
+                Second = person2
+            })
+            .Where(twoPerson => twoPerson.First != twoPerson.Second)
+            .Select(twoPersons =>
+            {
+                var listFirst = twoPersons.First.About.Split('.', '?', '!', ' ', ';', ':', ',');
+                var listSecond = twoPersons.Second.About.Split('.', '?', '!', ' ', ';', ':', ',');
+
+                return new
+                {
+                    FirstPerson = twoPersons.First,
+                    SecondPerson = twoPersons.Second,
+                    SimilarWordsInAbout = listFirst.Intersect(listSecond).Count()
+                };
+            }).ToList();
```

Comment:
not critical, but here you can skip converting to list because below you execute only single MaxBy